Feature/SWT-61 geometry

jennyknuth opened the pull request:
This PR allows the user to select various pre-loaded geometry files as well as upload their own .wrl files.

jennyknuth committed:
add dropdown with geometry file option group and custom file uploader

jennyknuth committed:
WIP first pass at file upload and image display feature

jennyknuth committed:
change geometry files from array of strings to array of objects

jennyknuth committed:
add /geometry and /image endpoints and ids

jennyknuth committed:
better spacing

jennyknuth committed:
rename and clean up

jennyknuth committed:
add .wrl format

jennyknuth committed:
remove unused vrml child

greglucas commented on src/app/containers/model/model.container.ts:
```diff
@@ -151,13 +183,40 @@ export class ModelComponent implements OnInit {
-                Object.keys(data).forEach( key => results[key] = this.round(data[key], 4));
+                Object.keys( data ).forEach( key => results[key] = this.round( data[key], 4 ));
                 this.results = results;
+                if ( this.imageFileId ) {
```
`&& this.vrmlImageSrc === null`

Right now, I think it is getting a new image on every submit because it clears the image every time. My thought would be to only get a new image if there isn't one already.

jennyknuth replied:
Now it only gets the image if the objectType, pitch, or sideslip has changed

greglucas commented on src/app/containers/model/model.container.html:
```diff
@@ -9,43 +9,63 @@
+                                            {{type === 'plate' ? 'plate (one sided)' : type}}
+                                        </mat-option>
+                                    </ng-container>
+                                    <mat-optgroup label="geometry files (.wrl format):">
```
When first opening up the dropdown menu I didn't realize there were other options below that I could scroll to. It stopped at SORCE.
Some ideas...

- Is there a way to force all of them to show up since we don't have that many?
- Make the scroll wheel permanently show up? (On my browser it auto-hides, which made this a bit tricky to know it was a scrollable field)
- Do a sub-field. So, when you go to the (currently gray) geometry files label, it breaks out to the right into another field selector just for the geometry files section.

jennyknuth replied:
It should be full height on most screens now

greglucas reviewed (approved):
Awesome, @jennyknuth! Looking real nice. I've just got a couple quick comments on it. But, it worked flawlessly for me :)

jennyknuth committed:
enlarge dropdown panel

jennyknuth committed:
only fetch image if objectType, pitch, sideslip changed; validate file upload

jennyknuth committed:
update npm packages

jennyknuth committed:
also add package-lock

jennyknuth commented:
I added validation that files were uploaded. Now if you select file upload and don't upload a file, submit is disabled and there is a message to choose a .wrl file. 

I also updated the npm packages, so I hope that fixes the security issues.

greglucas reviewed (approved):
Looks great, @jennyknuth!

The other thing I just noticed is that the first curly brace in the Submission display is really far over to the right. I'm not sure if that is how it is supposed to look or not. In fact, I'm not even sure we actually need that field at all anymore?

```javascript
                        {
  "objectType": "custom",
  "diameter": 1.212,
  "length": 2.01,
  "area": 3.4,
  "pitch": 80,
  "sideslip": 12.5,
  "temperature": 1200.5,
  "speed": 7800.45,
  "composition": {
    "o": 100000000000,
    "o2": 5000000,
    "n2": 1000000,
    "he": 1000000,
    "h": 10000
  },
  "accommodationModel": "SESAM",
  "energyAccommodation": 0.93,
  "surfaceMass": 65
}
```

Looks good to me though, feel free to deploy when you want.
